auth0/v3/management/rest.py

```diff
@@ -1,16 +1,87 @@
-import sys
-import json
 import base64
+import json
+import sys
+import time
+
 import requests
+
 from ..exceptions import Auth0Error
 
 
-class RestClient(object):
+class NopRateLimiter(object):
+    def update(self, headers):
+        pass
+
+    def ensure_limit(self):
+        pass
+
+
+class RateLimiter(object):
+    def __init__(self):
+        self.ratelimit_reset = int(time.time()) - 1
+        self.ratelimit_remaining = 1
+
+    def update(self, headers):
+        if 'x-ratelimit-remaining' in headers:
+            self.ratelimit_remaining = int(headers['x-ratelimit-remaining'])
+
+        if 'x-ratelimit-reset' in headers:
+            self.ratelimit_reset = int(headers['x-ratelimit-reset'])
+
+    def ensure_limit(self):
+        self.ratelimit_remaining -= 1
+
+        if self.ratelimit_remaining < 0:
+            sleep_time = self.ratelimit_reset - int(time.time())
+            if sleep_time > 0:
+                time.sleep(sleep_time)
+
+
+def extract_content(func):
+    def _decorator(self, *args, **kwargs):
+        response = func(self, *args, **kwargs)
```

Comment:
Isn't this already done [here at line 63](https://github.com/auth0/auth0-python/pull/83/files#diff-0bdc5dd5a292b9bc34852463b9e4cf2bR63)? I think you're calling this 2 times.

Reply:
No, line 63 calls the decorator which is returned here. Func is just a parameter of the decorator defining functions. Te decorators are cascaded in line 76.